From 68be611c53aa9b2c5004f43ec7e9b53b13673c84 Mon Sep 17 00:00:00 2001
From: aangerma <Avishag.Angerman@intel.com>
From: icarpis <itay.carpis@intel.com>
From: Evgeni Raikhel <evgeni.raikhel@intel.com>
Date: Thu, 30 Aug 2018 16:07:01 +0300

Subject: [PATCH] RS4xx, ZR300, RS5 Pixel formats patch
Signed-off-by: Evgeni Raikhel <evgeni.raikhel@intel.com>
---
 drivers/media/v4l2-core/v4l2-ioctl.c | 6 +++++
 1 files changed, 5 insertions(+)

diff --git a/4.4.0/drivers/media/v4l2-core/v4l2-ioctl.c b/4.4.0/drivers/media/v4l2-core/v4l2-ioctl.c
index 7486af2..124dd35 100644
--- a/4.4.0/drivers/media/v4l2-core/v4l2-ioctl.c
+++ b/4.4.0/drivers/media/v4l2-core/v4l2-ioctl.c
@@ -1229,6 +1229,16 @@ static void v4l_fill_fmtdesc(struct v4l2_fmtdesc *fmt)
 	case V4L2_SDR_FMT_CS8:		descr = "Complex S8"; break;
 	case V4L2_SDR_FMT_CS14LE:	descr = "Complex S14LE"; break;
 	case V4L2_SDR_FMT_RU12LE:	descr = "Real U12LE"; break;
+	case V4L2_PIX_FMT_Y8I:		descr = "8-bit Greyscale L/R interleaved"; break;
+	case V4L2_PIX_FMT_Y12I:		descr = "12-bit Grey L/R interleaved"; break;
+	case V4L2_PIX_FMT_Z16:		descr = "16-bit Depth data"; break;
+	case V4L2_PIX_FMT_RW16:		descr = "16-bit Raw data"; break;
+	case V4L2_PIX_FMT_INZI:		descr = "32-bit IR:Depth 10:16"; break;
+	case V4L2_PIX_FMT_W10:		descr = "10-bit packed 8888[2222]"; break;
+	case V4L2_PIX_FMT_CONFIDENCE_MAP:	descr = "4-bit per pixel packed"; break;
+	case V4L2_PIX_FMT_FG:		descr = "Frame Grabber (FG  )"; break;
+	case V4L2_PIX_FMT_INZC:		descr = "Planar Depth/Confidence (INZC)"; break;
+	case V4L2_PIX_FMT_PAIR:		descr = "Relative IR (PAIR)"; break;
 
 	default:
 		/* Compressed formats */
-- 
2.7.4

